modules/masterbar/admin-menu/class-admin-menu.php

```diff
@@ -0,0 +1,736 @@
+					site_is_coming_soon() ? esc_html__( 'Coming Soon', 'jetpack' ) : esc_html__( 'Private', 'jetpack' )
+				);
+			}
+		}
+
+		$site_card = '
+<div class="site__info">
+	<div class="site__title">%1$s</div>
+	<div class="site__domain">%2$s</div>
+	%3$s
+</div>';
+
+		$site_card = sprintf(
+			$site_card,
+			get_option( 'blogname' ),
+			$domain,
+			$badge
+		);
+
+		add_menu_page( 'site-card', $site_card, 'read', get_home_url(), null, $icon, 1 );
+		add_filter( 'add_menu_classes', array( $this, 'set_site_card_menu_class' ) );
+	}
+
+	/**
+	 * Adds a custom element class and id for Site Card's menu item.
+	 *
+	 * @param array $menu Associative array of administration menu items.
+	 * @return array
+	 */
+	public function set_site_card_menu_class( array $menu ) {
+		foreach ( $menu as $key => $menu_item ) {
+			if ( 'site-card' !== $menu_item[3] ) {
+				continue;
+			}
+
+			$classes = ' toplevel_page_site-card';
+
+			// webclip.png is the default on WoA sites. Anything other than that means we have a custom site icon.
+			if (
+				( function_exists( 'blavatar_exists' ) && blavatar_exists( ( new Status() )->get_site_suffix() ) ) ||
+				( has_site_icon() && 'https://s0.wp.com/i/webclip.png' !== get_site_icon_url( 512 ) )
+			) {
+				$classes .= ' has-site-icon';
+			}
+
+			$menu[ $key ][4] = $menu_item[4] . $classes;
+			$menu[ $key ][5] = 'toplevel_page_site_card';
+			break;
+		}
+
+		return $menu;
+	}
+
+	/**
+	 * Adds My Home menu.
+	 *
+	 * @param string $domain  Site domain.
+	 * @param bool   $calypso Optional. Whether links should point to Calypso or wp-admin. Default true (Calypso).
+	 */
+	public function add_my_home_menu( $domain, $calypso = true ) {
+		global $submenu;
+
+		$menu_slug = $calypso ? 'https://wordpress.com/home/' . $domain : 'index.php';
+
+		remove_menu_page( 'index.php' );
+		remove_submenu_page( 'index.php', 'index.php' );
+
+		add_menu_page( __( 'My Home', 'jetpack' ), __( 'My Home', 'jetpack' ), 'read', $menu_slug, null, 'dashicons-admin-home', 2 );
+
+		// Only add submenu when there are other submenu items.
+		if ( ! empty( $submenu['index.php'] ) ) {
+			add_submenu_page( $menu_slug, __( 'My Home', 'jetpack' ), __( 'My Home', 'jetpack' ), 'read', $menu_slug, null, 1 );
+		}
+
+		$this->migrate_submenus( 'index.php', $menu_slug );
+	}
+
+	/**
+	 * Adds Stats menu.
+	 *
+	 * @param string $domain Site domain.
+	 */
+	public function add_stats_menu( $domain ) {
+		$menu_title = __( 'Stats', 'jetpack' );
+
+		if ( $this->is_wpcom_site() && ! $this->is_api_request ) {
+			$menu_title .= sprintf(
+				'<img class="sidebar-unified__sparkline" width="80" height="20" src="%1$s" alt="%2$s">',
+				esc_url( home_url( 'wp-includes/charts/admin-bar-hours-scale-2x.php?masterbar=1&s=' . get_current_blog_id() ) ),
+				esc_attr__( 'Hourly views', 'jetpack' )
+			);
+		}
+
+		add_menu_page( __( 'Stats', 'jetpack' ), $menu_title, 'edit_posts', 'https://wordpress.com/stats/day/' . $domain, null, 'dashicons-chart-bar', 3 );
```

Comment:
On a similar note, this gets a bit confusing if you also use WooCommerce. We now have two top-menu stats:

![image](https://user-images.githubusercontent.com/426388/100646681-9e320800-333e-11eb-940f-74b8f4f59560.png)

Reply:
> This gives us 2 different Stats menu items now. Should we remove the one that's in the Jetpack menu?

The Jetpack menu and what should become of some of the links is a conversation we still have to have. Including where it should live. Site stats and Akismet are probably the two that I'd suggest removing. The proposed menu structure (see  27579-pb) has the Jetpack menu also living between Comments/Feedback and Appearance.

> On a similar note, this gets a bit confusing if you also use WooCommerce. We now have two top-menu stats
@lessbloat Situations like this ^ are probably to be expected now with plugins adding menu items. Would you want to rename "Stats" or keep it as is?

Reply:
> The Jetpack menu and what should become of some of the links is a conversation we still have to have.

I feel like this is more up to the JP team to decide. cc @folletto 

> this gets a bit confusing if you also use WooCommerce. We now have two top-menu stats

Again, I feel like this is up to the Woo team to decide how they want to handle it. @josemarques

Reply:
I feel a bit uncomfortable reviewing this inside a comment thread on a line on a PR 😅 because I agree that yes, this is a larger conversation. 

Is there already a place somewhere -or- this is it, and do we need to kick that off?

Reply:
Heh. Fair point! I'm happy to move this to P2. I'll kick off a separate convo with additional context and cc you and a few others for feedback.

Reply:
Thank you 💙

Reply:
> Again, I feel like this is up to the Woo team to decide how they want to handle it. @josemarques

@davemart-in Sorry for getting here late. This might be fixed with the introduction on the new Woo Navigation since everything will be under the WooCommerce folder, but we might need an interim solution. Pinging @pmcpinto and @elizaan36 to take a look here and find a solution.

Reply:
We plan to roll out the new WooCommerce navigation for eComm users next week and for Business plan users in mid-January, so I think we don't need to spend time/resources on an interim solution.